### ISSUE TITLE
The "Open full report" button might resize awkwardly

### ISSUE THREAD

**Issue by guihkx** [GyulyVGC/sniffnet#48]
**Describe the bug**
Sometimes, IPv6 addresses pop up on the list, and if the main window doesn't have enough room horizontally (e.g., when Sniffnet is not maximized), the "Open full report" button gets resized improperly, and the text inside of the button gets removed.

**To Reproduce**
1. Unmaximize Sniffnet's window
2. Resize the window to its lowest possible horizontal size
3. Wait for an IPv6 address to show up on the list
4. Notice how the "Open full report" button gets incorrectly resized

**Expected behavior**
The "Open full report" button should not be resized. Probably.

**Screenshots**

https://user-images.githubusercontent.com/626206/209119297-136b77ed-9916-43e7-bec7-df7ac64aa552.mp4

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Version: 1.0.1

**Comment by GyulyVGC** [comment]
Thanks for having spotted this issue.
Actually not all the IPv6 addresses cause this problem but only that longer than 25 characters, because that is the maximum length currently possible for that field. I thought it to be enough in the most cases since I've never received something longer, but apparently your local interface has a longer address.

Could you also send a screenshot about the file with the report? For the file I've considered this case so it should be fine, but having a confirmation would be great.

**Comment by guihkx** [comment]
Sure:

```
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
|     Src IP address      | Src port |     Dst IP address      | Dst port | Layer 4 | Layer 7 |   Packets  |   Bytes    |  Initial timestamp  |   Final timestamp   |
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
|     155.133.249.197     |   27019  |       192.168.1.2       |   56986  |   TCP   |  Other  |         1  |     32  B  | 2022-12-22 08:07:10 | 2022-12-22 08:07:10 |                                        
|     162.254.196.84      |   27018  |       192.168.1.2       |   54442  |   TCP   |  Other  |         2  |     40  B  | 2022-12-22 08:07:10 | 2022-12-22 08:07:10 |                                        
|     162.254.196.84      |   27018  |       192.168.1.2       |   45948  |   TCP   |  Other  |         2  |     40  B  | 2022-12-22 08:07:10 | 2022-12-22 08:07:10 |                                        
|       192.168.1.2       |   33468  |     155.133.249.197     |   27019  |   TCP   |  Other  |         2  |    148  B  | 2022-12-22 08:07:11 | 2022-12-22 08:07:14 |                                        
|   2804:14d:9083:8390:d03d:3eab:60ff:ea94    |   59898  |     2a03:2880:f205:2c5:face:b00c:0:167      |     443  |   TCP   |  HTTPS  |         2  |    133  B  | 2022-12-22 08:07:11 | 2022-12-22 08:07:11 |
|     2a03:2880:f205:2c5:face:b00c:0:167      |     443  |   2804:14d:9083:8390:d03d:3eab:60ff:ea94    |   59898  |   TCP   |  HTTPS  |         2  |    135  B  | 2022-12-22 08:07:11 | 2022-12-22 08:07:11 |
|       192.168.1.2       |   52002  |      190.217.33.68      |   27017  |   TCP   |  Other  |         2  |    120  B  | 2022-12-22 08:07:11 | 2022-12-22 08:07:11 |                                        
|     155.133.249.197     |   27019  |       192.168.1.2       |   33468  |   TCP   |  Other  |         2  |     64  B  | 2022-12-22 08:07:11 | 2022-12-22 08:07:14 |                                        
```

**Comment by GyulyVGC** [comment]
As I expected. 
The alignment is screwed up because I kept the column length the same even in that case, but the following lines seem to not be affected, because in case addresses are shorter I added padding spaces at the end of the line.
This is particularly crucial because the file is updated every second through seek operations to the changed lines.

I considered this case an exceptional scenario, but I'll think I have to redesign it.
The fact is that I feel like it'd be not-so-nice to have such long fields even if addresses are a lot shorter.

For which concerns the GUI, I think I will truncate it because there is not enough physical space; for which concerns the file I will think to a better solution. 

In the meantime, every suggestion is appreciated.

**Comment by guihkx** [comment]
>For which concerns the GUI, I think I will truncate it because there is not enough physical space

I think that should be fine. Does `iced` have something like tooltips for text? If it has, maybe an improvement would be to display the non-truncated IP address in a tooltip, upon hovering the mouse cursor on it.

**Comment by GyulyVGC** [comment]
> >For which concerns the GUI, I think I will truncate it because there is not enough physical space
> 
> 
> 
> I think that should be fine. Does `iced` have something like tooltips for text? If it has, maybe an improvement would be to display the non-truncated IP address in a tooltip, upon hovering the mouse cursor on it.

That'd be really nice. I have to investigate if iced offer such functionalities.

I'm also implementing a deeper inspection page for each connection for the next release.
Every address will have more space and thus at least in that page I'm sure I'll be able to display the longest addresses without any problem.

**Comment by GyulyVGC** [comment]
Since this issue was related to the resizing of the button, I consider it solved.
I've added a limit to the dimension of the report row and I embedded it in an horizontal scrollable.

![Screenshot 2023-02-08 at 08 35 38](https://user-images.githubusercontent.com/100347457/217465045-864f0eef-7f2b-4361-b032-631514d60835.png)

The dealignment of text fields still happens and I've just opened a new issue for it (see #63).
